lib/helpers/firebase_functions.dart

```diff
@@ -209,73 +161,136 @@ class FirebaseFunctions {
-        .where("userId", isEqualTo: auth.currentUser!.uid)
-        .get();
+  Future availabilityMultiPathUpdate({required String itemKey, required bool currentValue}) async {
+    final QuerySnapshot categoriesSnapshot = await firestore.collection('categories').where("userId", isEqualTo: auth.currentUser!.uid).get();
 
     if (categoriesSnapshot.size == 0) {
       return throw FirebaseException(plugin: "User has no categories");
     }
 
     for (final DocumentSnapshot category in categoriesSnapshot.docs) {
-      final QuerySnapshot categoryItemsSnapshot = await firestore
-          .collection('categoryItems/${category.id}/items')
-          .where(FieldPath.documentId, isEqualTo: itemKey)
-          .get();
+      final QuerySnapshot categoryItemsSnapshot = await firestore.collection('categoryItems/${category.id}/items').where(FieldPath.documentId, isEqualTo: itemKey).get();
 
       for (final DocumentSnapshot item in categoryItemsSnapshot.docs) {
-        final DocumentReference itemReference = firestore
-            .collection('categoryItems/${category.id}/items')
-            .doc(item.id);
+        final DocumentReference itemReference = firestore.collection('categoryItems/${category.id}/items').doc(item.id);
 
         await itemReference.update({"is_available": !currentValue});
       }
     }
   }
 
-  /// First, the method updates the availability status of the item [itemId] in the 'items' collection, 
+  /// First, the method updates the availability status of the item [itemId] in the 'items' collection,
   /// then, if the operation is successful, it calls availabilityMultiPathUpdate method.
   /// If not, it returns boolean false.
   Future updateItemAvailability({required String itemId}) async {
     try {
-      final DocumentReference itemRef =
-          firestore.collection("items").doc(itemId);
+      final DocumentReference itemRef = firestore.collection("items").doc(itemId);
       final DocumentSnapshot documentSnapshot = await itemRef.get();
-      final Map<String, dynamic> data =
-          documentSnapshot.data() as Map<String, dynamic>;
+      final Map<String, dynamic> data = documentSnapshot.data() as Map<String, dynamic>;
       final bool? currentValue = data["is_available"];
       // Items collection update
-      await firestore
-          .collection("items")
-          .doc(itemId)
-          .update({"is_available": !currentValue!}).then(
-        (_) => availabilityMultiPathUpdate(
-            itemKey: itemId, currentValue: currentValue),
+      await firestore.collection("items").doc(itemId).update({"is_available": !currentValue!}).then(
+        (_) => availabilityMultiPathUpdate(itemKey: itemId, currentValue: currentValue),
       );
     } catch (e) {
       return false;
     }
     return true;
   }
+
+  /// Delete category document from categories collection
+  /// Delete associated categoryItems document
+  Future deleteCategory({required String categoryId}) async {
```

Comment:
Why is this a nested method?

Reply:
Because it's a private function only used within the `deleteCategory()` function. While this could be written without nesting (having the code from `deleteFromCollection()` written simply within its parent function, nesting helps with readability and code reuse, all while keeping it private (it need not be accessed from elsewhere)